checklist_plus/text_generation/llm.py

```
@@ -0,0 +1,688 @@
+            formatted_prompts.append(formatted_prompt)
+
+        try:
+            # Use structured output with Pydantic model
+            structured_llm = self.llm_client.with_structured_output(UniqueCompletions)
+            # print("here")
+            responses = structured_llm.batch(formatted_prompts)
+            # print("responses:", responses)
+
+            # Extract completions from Pydantic model
+            completions_all = [resp.completions for resp in responses if hasattr(resp, 'completions')]
+            assert len(completions_all) == len(formatted_prompts), "Mismatch in number of responses"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)